tests/object.rs

```diff
@@ -188,3 +187,31 @@ fn f32_nii_gz() {
     assert_eq!(volume.get_f32(&[5, 0, 4]).unwrap(), 0.4);
     assert_eq!(volume.get_f32(&[0, 8, 5]).unwrap(), 0.8);
 }
+
+#[cfg(feature = "ndarray_volumes")]
```

Comment:
Since this new capability isn't restricted to ndarray volumes, how about we make it happen without "ndarray_volumes"? The last part can be kept behind a `cfg!` condition:

```rust
if cfg!(feature = "ndarray_volume") {
    use ndarray::Ix3;

    let dyn_data = volume.to_ndarray::<f32>().unwrap();
    ...
}
```

Reply:
I don't want to use you as a debugger but it's not working.

    // Missing crate ndarray, no IntoNdArray in the root, etc.
    if cfg!(feature = "ndarray_volumes") {
        use ndarray::Ix3;
        use nifti:: IntoNdArray;
        ...

    // Compiles and runs
    #[cfg(feature = "ndarray_volumes")] {
        use ndarray::Ix3;
        ...

I can't find any proof, but it looks like it can only be used for simple condition. Have you used it with `extern crate` and/or `use` before?

Reply:
Well, it seems that you are right. Conditional expressions based on the outcome of `cfg!` do not prevent their respective blocks from being analysed, even when that outcome is known at compile time. We have to employ directives, as you suggested.

```diff
:100644 100644 49128d1 0000000 M	tests/object.rs

diff --git a/tests/object.rs b/tests/object.rs
index 49128d1..544ea14 100644
--- a/tests/object.rs
+++ b/tests/object.rs
@@ -188,11 +188,9 @@ fn f32_nii_gz() {
     assert_eq!(volume.get_f32(&[0, 8, 5]).unwrap(), 0.8);
 }
 
-#[cfg(feature = "ndarray_volumes")]
 #[test]
 fn test_false_4d() {
-    use ndarray::Ix3;
-    use nifti:: {InMemNiftiVolume, IntoNdArray};
+    use nifti::InMemNiftiVolume;
 
     let (w, h, d) = (5, 5, 5);
     let mut header = NiftiHeader {
@@ -210,8 +208,15 @@ fn test_false_4d() {
         volume = InMemNiftiVolume::from_raw_data(&header, volume.to_raw_data()).unwrap();
     }
     assert_eq!(volume.dimensionality(), 3);
-    let dyn_data = volume.to_ndarray::<f32>().unwrap();
-    assert_eq!(dyn_data.ndim(), 3);
-    let data = dyn_data.into_dimensionality::<Ix3>().unwrap();
-    assert_eq!(data.ndim(), 3); // Obvious, but it's to avoid being optimized away
+
+    #[cfg(feature = "ndarray_volumes")]
+    {
+        use ndarray::Ix3;
+        use nifti::IntoNdArray;
+
+        let dyn_data = volume.to_ndarray::<f32>().unwrap();
+        assert_eq!(dyn_data.ndim(), 3);
+        let data = dyn_data.into_dimensionality::<Ix3>().unwrap();
+        assert_eq!(data.ndim(), 3); // Obvious, but it's to avoid being optimized away
+    }
 }
```

But now I also noticed that this is in tests/object.rs. I would understand this as a unit test for the in-memory nifti volume rather than an integration test for a nifti object. Can you move it to the `test` module in "src/volume/inmem.rs"?